test/index.ts

```diff
@@ -189,6 +192,127 @@ describe('Storage', () => {
     });
   });
 
+  describe('createHmacKey', () => {
+    const SERVICE_ACCOUNT_EMAIL = 'service-account@gserviceaccount.com';
+    const ACCESS_ID = 'some-access-id';
+    const metadataResponse = {
+      accessId: ACCESS_ID,
+      etag: 'etag',
+      id: ACCESS_ID,
+      projectId: 'project-id',
+      serviceAccountEmail: SERVICE_ACCOUNT_EMAIL,
+      state: 'ACTIVE',
+      timeCreated: '20190101T00:00:00Z',
+      updated: '20190101T00:00:00Z',
+    };
+    const response = {
+      secret: 'my-secret',
+      metadata: metadataResponse,
+    };
+
+    it('should make correct API request', done => {
+      storage.request = (
+        reqOpts: DecorateRequestOptions,
+        callback: Function
+      ) => {
+        assert.strictEqual(reqOpts.method, 'POST');
+        assert.strictEqual(
+          reqOpts.uri,
+          `/projects/${storage.projectId}/hmacKeys`
+        );
+        assert.strictEqual(
+          reqOpts.qs.serviceAccountEmail,
+          SERVICE_ACCOUNT_EMAIL
+        );
+
+        callback(null, response);
+      };
+
+      storage.createHmacKey(SERVICE_ACCOUNT_EMAIL, done);
+    });
+
+    it('should throw without a serviceAccountEmail', () => {
+      assert.throws(
+        () => storage.createHmacKey(),
+        /must be a service account email/
+      );
+    });
+
+    it('should throw when first argument is not a string', () => {
+      assert.throws(
+        () =>
+          storage.createHmacKey({
+            userProject: 'my-project',
+          }),
+        /must be a service account email/
+      );
+    });
+
+    it('should honor the userProject option', async () => {
```

Comment:
Since we don't have any special lines in the code that handle this option, I don't think we should have a test for it. If you want to just make sure that arbitrary user-provided options are treated as a query object (the line `const query = Object.assign({}, options, {serviceAccountEmail});`), let's have a test for that. (You could also test that it doesn't modify the user's object as well)

Reply:
536c8aa